### ISSUE TITLE
Deterministic Nature of mlls.exact_marginal_log_likelihood

### ISSUE THREAD

**Issue by sberglin** [cornellius-gp/gpytorch#972]
# 📚 Documentation/Examples

I have a question about the nature of how `mlls.exact_marginal_log_likelihood` is computed. My understanding is that it computes the marginal likelihood of model predictions given data as a "loss." However, repeatedly computing it on the full training set results in different values and gradients, with parameters set.

I have attached some [code](https://github.com/cornellius-gp/gpytorch/files/3888090/reproduce.txt) to reproduce the behavior. I am curious why this is happening, or if I am making an error. If this is expected behavior, some added documentation [here](https://gpytorch.readthedocs.io/en/latest/marginal_log_likelihoods.html) may be useful for other users!

**Comment by gpleiss** [comment]
Hi @sberglin - when N is sufficiently large we use a stochastic unbiased estimate for the log determinant term in the marginal log likelihood. For more details, please see our NeurIPS 2018 paper (https://arxiv.org/abs/1809.11165)

**Comment by gpleiss** [comment]
You can make the computation deterministic by using the context manager `with gpytorch.settings.max_cholesky_size(<num_data_points>)`, where `num_data_points` should be equal to or larger than N. This forces GPyTorch to use the cholesky decomposition to compute the MLL, and this computation will be deterministic.

We're a bit swamped right now improving other parts of the documentation, so if you have suggestions for where to add docs a PR would be very welcome!

**Comment by sberglin** [comment]
Got it! I tried it out and it behaves as expected. This includes making the gradient computations deterministic as well. I'll throw up a PR. Thank you very much!
